ffi/ffi_wrapper.go

```diff
@@ -56,41 +289,81 @@ func CreateTwinDidWithControlDelegation(
 	}
 	twinIdentity, err := api.CreateTwinWithControlDelegation(resolver, opts)
 	if err != nil {
-		return C.CString(""), C.CString(fmt.Sprintf("FFI lib error: creating twin failed: %+vs", err))
+		return nil, C.CString(fmt.Sprintf("FFI lib error: creating twin DiD Doc failed: %+vs", err))
 	}
-	return C.CString(twinIdentity.Did()), C.CString("")
+	return C.CString(twinIdentity.Did()), nil
 }
 
 //export FreeUpCString
 func FreeUpCString(pointer *C.char) {
 	C.free(unsafe.Pointer(pointer))
 }
 
-func getAgentIdentity(
-	agentDid string, agentKeyName string, agentName string, seed string,
-) (register.RegisteredIdentity, []byte, error) {
+func createIdentity(isUser bool, // true for userId, false for agentId
+	cResolverAddress *C.char, cKeyName *C.char, cName *C.char, cSeed *C.char,
+) (*C.char, *C.char) {
+	var err error
+	resolverAddress := C.GoString(cResolverAddress)
+	keyName := C.GoString(cKeyName)
+	name := C.GoString(cName)
+	seed := C.GoString(cSeed)
+
+	seedBytes, err := hex.DecodeString(seed)
+	if err != nil {
+		return nil, C.CString(fmt.Sprintf("FFI lib error: failed to decode seed: %+v", err))
+	}
+	addr, err := url.Parse(resolverAddress)
+	if err != nil {
+		return nil, C.CString(fmt.Sprintf("FFI lib error: parsing resolver address failed: %+v", err))
+	}
+
+	opts := &api.CreateIdentityOpts{
+		Seed:    seedBytes,
+		KeyName: keyName,
+		//Password: nil,
```

Comment:
Is this a TODO?

Reply:
it should not be a todo and password should be passed as an argument. future enhancement